packages/common/src/utils/submittedDataUtils.ts

```diff
@@ -0,0 +1,85 @@
+import { functions } from '@overturebio-stack/lectern-client';
+import {
+	SchemaData,
+	SchemaValidationError,
+	SchemasDictionary,
+} from '@overturebio-stack/lectern-client/lib/schema-entities.js';
+import { groupBy, has } from 'lodash-es';
+import { Dependencies } from '../config/config.js';
+import { NewSubmittedData } from '../models/submitted_data.js';
+
+const utils = (dependencies: Dependencies) => {
+	const LOG_MODULE = 'SUBMITTED_DATA_UTILS';
+	const { logger } = dependencies;
+	return {
+		/**
+		 * Creates a map of SubmittedData grouped by entities
+		 * @param {Array<NewSubmittedData>} data
+		 * @returns
+		 */
+		mapSchemaDataByEntity: (
+			data: Array<NewSubmittedData>,
+		): { original: Record<string, NewSubmittedData[]>; schemaData: Record<string, SchemaData> } => {
+			return data.reduce(
+				(acc: { original: Record<string, NewSubmittedData[]>; schemaData: Record<string, SchemaData> }, cur) => {
+					acc.schemaData[cur.entityName] = [...(acc.schemaData[cur.entityName] || []), { ...cur.data }];
+					acc.original[cur.entityName] = [...(acc.original[cur.entityName] || []), { ...cur }];
+					return acc;
+				},
+				{ original: {}, schemaData: {} },
```

Comment:
we need to keep a 1:1 relation between the original data with the parsed data. 
This function only converts the submission Data from one type to another (Record<string, NewSubmittedData[]> -> Record<string, SchemaData>)

Reply:
I agree that `original` is not clear. it's a list of `submittedEntities` I think?

Reply:
Think I'm struggling with the semantics a little. Hopefully others will find this more declarative than I, but I now see Jon agrees...
Perhaps something like `sourceData` (instead of `original`), and `dataByEntityName` (instead of `schemaData`) would be more descriptive

Reply:
1. the `original` field is an array of SubmittedData (this is the record original from the database meaning it contains other fields like ids, timestamps, etc not relevant for the dictionary validation). this lists of SubmittedData is grouped by entityName as a map/Record. 
2. `schemaData` is a list of only the data we need for the dictionary validation. returned as a Record compatible for the lectern-client.

Having the `original` records from the database and it's `schemaData` is useful to match each validation result and update the record on the database accordingly.

Reply:
changed the variable names of this function to:
- `original` changed to `submittedDataByEntityName`
- `schemaData` changed to `schemaDataByEntityName `

Reply:
better being over verbose, than under 👍 
thank you